presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -229,10 +238,34 @@ public void start()
             }, 1, 1, MILLISECONDS);
             if (isResourceManagerEnabled) {
                 resourceGroupRuntimeExecutor.start();
+                resourceGroupHeartbeatExecutor.start();
             }
         }
     }
 
+    private void sendResourceGroupHeartbeat()
+    {
+        ImmutableList.Builder<ResourceGroupRuntimeInfo> resourceGroupRuntimeInfos = ImmutableList.builder();
+        rootGroups.forEach(resourceGroup -> buildResourceGroupRuntimeInfo(resourceGroupRuntimeInfos, resourceGroup));
+        List<ResourceGroupRuntimeInfo> finalResourceGroupRuntimeInfos = resourceGroupRuntimeInfos.build();
+        if (finalResourceGroupRuntimeInfos.size() > 0) {
+            clusterStatusSender.sendResourceGroupRuntimeHeartbeat(finalResourceGroupRuntimeInfos);
+        }
+    }
+
+    private void buildResourceGroupRuntimeInfo(ImmutableList.Builder<ResourceGroupRuntimeInfo> resourceGroupRuntimeInfos, InternalResourceGroup resourceGroup)
+    {
+        if (resourceGroup.subGroups().isEmpty()) {
+            ResourceGroupRuntimeInfo.Builder resourceGroupRuntimeInfo = ResourceGroupRuntimeInfo.builder(resourceGroup.getId());
+            resourceGroupRuntimeInfo.addRunningQueries(resourceGroup.getRunningQueries());
+            resourceGroupRuntimeInfo.addQueuedQueries(resourceGroup.getQueuedQueries());
```

Comment:
One optimization can be done here is: If there is no running queries/queued queries then don't build the runtime info. This helps reduce the number of resource group runtime info we send from coordinator to resource manager.